### PR TITLE
[9.0][MIG] account_invoice_currency

### PR DESCRIPTION
**Pull request by antespi**
- Reuse account v9 fields
- Adds field not found in account v9 (amount_tax_signed)
- Guidelines complaince

### CODE REVIEW

File: account_invoice_currency/README.rst [hunk 1]
```diff
@@ -7,13 +12,39 @@ amount untaxed, amount taxed and amount total.
...
+**NOTE #2**: Please note that these fields are negative (as done in v9 account addon)
+for credit notes (for example a refund invoice)
+
+
+Known issues / Roadmap
+======================
+
+These fields name change from v8:
+
+* cc_amount_untaxed -> amount_untaxed_signed (already in account v9)
```
Comment (pedrobaeza) [inline account_invoice_currency/README.rst:30]: Please prepare a migration script (using openupgradelib) that renames the columns and invert sign of the corresponding invoices.

File: account_invoice_currency/README.rst [hunk 2]
```diff
@@ -25,16 +56,20 @@ Contributors
 * Jordi Esteve <jesteve@zikzakmedia.com>
 * Joaquín Gutierrez <joaquing.pedrosa@gmail.com>
 * Pedro M. Baeza <pedro.baeza@serviciosbaeza.com>
+* Antonio Espinosa <antonioea@antiun.com>
```
Comment (pedrobaeza) [inline account_invoice_currency/README.rst]: Why your old Antiun account? You should remove it and leave only the current one, as you are no more active there. The trace of the company remains in authors, not here.

File: account_invoice_currency/__openerp__.py [hunk 3]
```diff
@@ -1,41 +1,29 @@
...
-    'version': "8.0.1.0.0",
-    'author': "Zikzakmedia SL, ,Odoo Community Association (OCA)"
+    'version': "9.0.1.0.0",
+    'author': "Zikzakmedia SL, "
+              "Odoo Community Association (OCA), "
               "Joaquín Gutierrez, "
-              "Serv. Tecnol. Avanzados - Pedro M. Baeza",
+              "Serv. Tecnol. Avanzados - Pedro M. Baeza, "
+              "Tecnativa",
     'website': "http://www.zikzakmedia.com, "
```
Comment (pedrobaeza) [inline account_invoice_currency/__openerp__.py]: This doesn't work, as the link is unique. Change it with the github URL of the project.

File: account_invoice_currency/models/account_invoice.py [hunk 4]
```diff
@@ -1,77 +1,28 @@
...
-                    if line.account_id.id in tax_accounts:
-                        self.cc_amount_tax += line.debit - line.credit
-                if self.type in ('out_invoice', 'in_refund'):
-                    self.cc_amount_untaxed = -self.cc_amount_untaxed
-                    self.cc_amount_tax = -self.cc_amount_tax
-                self.cc_amount_total = (self.cc_amount_tax +
-                                        self.cc_amount_untaxed)
+    @api.depends('invoice_line_ids.price_subtotal', 'tax_line_ids.amount',
+                 'currency_id', 'company_id')
+    def _compute_amount(self):
```
Comment (pedrobaeza) [inline account_invoice_currency/models/account_invoice.py:24]: Put after the field definition (as guidelines say).

**Comment by antespi** [comment]
OpenUpgrade script made

**Comment by antespi** [comment]
Any news here? We need this addon in order to migrate other addons at l10n-spain repository. Thanks all

**Comment by sergio-teruel** [comment]
@antespi make rebase.

**Comment by antespi** [comment]
Rebase done, thanks

File: account_invoice_currency/__init__.py [hunk 5]
```diff
@@ -1,20 +1,9 @@
...
-#
-# You should have received a copy of the GNU Affero General Public License
-# along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# © 2004-2011 Pexego Sistemas Informáticos. (http://pexego.es)
+# © 2004-2011 Zikzakmedia S.L. (http://zikzakmedia.com)
+#             Jordi Esteve <jesteve@zikzakmedia.com>
+# © 2014-2015 Serv. Tecnol. Avanzados - Pedro M. Baeza
+# © 2016 Antiun Ingenieria S.L. - Antonio Espinosa
```
Comment (pedrobaeza) [inline account_invoice_currency/__init__.py:6]: Why Antiun? I even prefer to remove copyright on this "dummy" fields

File: account_invoice_currency/i18n/es.po [hunk 6]
```diff
@@ -24,6 +24,42 @@ msgid "Amounts in company currency"
 msgstr "Importes moneda compañía"
 
 #. module: account_invoice_currency
+#: field:account.invoice,cc_amount_tax:0
+msgid "Company Cur. Tax"
+msgstr "Company Cur. Tax"
```
Comment (pedrobaeza) [inline account_invoice_currency/i18n/es.po]: Translate this

File: account_invoice_currency/i18n/pt_PT.po [hunk 7]
```diff
@@ -0,0 +1,65 @@
+# Translation of Odoo Server.
```
Comment (pedrobaeza) [inline account_invoice_currency/i18n/pt_PT.po]: Remove this empty one

File: account_invoice_currency/models/__init__.py [hunk 8]
```diff
@@ -1,20 +1,9 @@
...
-# This program is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU Affero General Public License for more details.
-#
-# You should have received a copy of the GNU Affero General Public License
-# along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# © 2004-2011 Pexego Sistemas Informáticos. (http://pexego.es)
```
Comment (pedrobaeza) [inline account_invoice_currency/models/__init__.py]: Remove copyrights here

**Comment by pedrobaeza** [comment]
cc @Tecnativa

**Comment by pedrobaeza** [comment]
@antespi, please take into account last remarks.

**Review by angelmoya — APPROVED** [review]
Test local and works fine.

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Please squash commit together by author and I'll merge.

**Comment by antespi** [comment]
Squash done

**Comment by gustavovalverde** [comment]
Is odoo using another approach in v10 for `amount_tax_signed`?

**Comment by pedrobaeza** [comment]
I don't know, but this is not definitively the place to ask for.

**Pull request merged**